### PR TITLE
Create graphql Plugin

### PR DESCRIPTION
**Pull request by joshmedeski**
Adds a package for the graphql language.

- Ensures the graphql language is installed (via nvim-treesitter)
- Ensures the graphql-language-service-cli LSP is installed (via Mason)
- Configures the graphql language server to work with "graphql", "typescript", and "typescriptreact" filetypes

### CODE REVIEW

**Comment by joshmedeski** [comment]
My bad, trying to sync my fork up to my upstream branch didn't work correctly. I'll work to fix it and push the changes.

**Comment by joshmedeski** [comment]
All done, ready for a review on this @folke

File: lua/lazyvim/plugins/extras/lang/graphql.lua [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      end
+    end,
+  },
+
+  -- correctly install lsp with mason
+  {
+    "williamboman/mason.nvim",
+    opts = function(_, opts)
+      vim.list_extend(opts.ensure_installed, {
+        "graphql-language-service-cli",
```
Comment (Jomik) [inline lua/lazyvim/plugins/extras/lang/graphql.lua:17]: Why is this needed? It wasn't needed for min 🤔 
https://github.com/Jomik/dotfiles/blob/7f709d13ae9676c76fb0ab5fa479001358ae7140/.config/nvim/lua/plugins/lang/graphql.lua#L14

File: lua/lazyvim/plugins/extras/lang/graphql.lua [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      })
+    end,
+  },
```
Comment (Jomik) [inline lua/lazyvim/plugins/extras/lang/graphql.lua:20]:
```suggestion
```
This isn't needed, `mason` installs the server already, since you put `graphql` in the `nvim-lspconfig` opts.
Here is my working config:
https://github.com/Jomik/dotfiles/blob/7f709d13ae9676c76fb0ab5fa479001358ae7140/.config/nvim/lua/plugins/lang/graphql.lua#L14

**Review by Jomik — CHANGES_REQUESTED** [review]

File: lua/lazyvim/plugins/extras/lang/graphql.lua (hunk 2 above)
Comment (joshmedeski) [reply lua/lazyvim/plugins/extras/lang/graphql.lua:20]: This plugin is meant to fully work without you having to add anything to your config other than the plugin line in `lazy.lua`.

Comment (Jomik) [reply lua/lazyvim/plugins/extras/lang/graphql.lua:20]: Yup.. There is no need to do anything more. `mason` automatically installs servers added in `nvim-lspconfig`.
That is done in all the other language plugins as well.
Check the json or go plugins for further examples.
https://github.com/LazyVim/LazyVim/blob/274649e92321d615888f889476c9329f8d831e8e/lua/lazyvim/plugins/extras/lang/json.lua#L23 
You can also simply try without and see if it isn't automatically installing it for you already 👍

Comment (joshmedeski) [reply lua/lazyvim/plugins/extras/lang/graphql.lua:10]: It does appear to still install when dropping the ensure install. Thanks

**Comment by joshmedeski** [comment]
@folke any change this will get merged soon?

**Comment by joshmedeski** [comment]
Closing due to no response from @folke.

Feel free to reopen this if you want to merge it.
